src/main/java/com/example/springassignment3/controller/PatientController.java

```diff
@@ -15,36 +18,58 @@
     @Autowired
     PatientService thePatientService;
 
+    Logger logger = LoggerFactory.getLogger(PatientController.class);
+
     @GetMapping("/patients")
     public List<Patient> getPatientList()
     {
+        logger.info("...Getting the list of all patients...");
         return thePatientService.findAllPatients();
     }
 
     @GetMapping("/patients/{patientId}")
     public Optional<Patient> getPatientById(@PathVariable int patientId)
     {
-        return thePatientService.findPatientById(patientId);
+        logger.info("...Getting patient based on the given Id...");
+        Optional<Patient> thePatient = thePatientService.findPatientById(patientId);
+
+        if(!thePatient.isPresent())
+        {
+            throw new PatientNotFoundException("Patient with id " + patientId + " could not be found");
+        }
+
+        return thePatient;
     }
 
     @PostMapping("/patients")
     public Patient savePatient(@RequestBody Patient thePatient)
     {
         thePatient.setId(0);
```

Comment:
In Patient entity id is autoincrement. So you should not manually set id anywhere